integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-spark-service-bundle/nifi-spark-service/src/main/java/com/thinkbiganalytics/nifi/v2/spark/SparkJobserverService.java

```diff
@@ -0,0 +1,357 @@
+        }
+        return success;
+    }
+
+    /**
+     * Using a timer task to periodically delete Spark contexts which have timed out
+     */
+    private void setupContextTimeoutTask() {
+
+        getLogger().info("Creating context timeout task");
+        TimerTask contextTimeoutTask = new TimerTask() {
+            @Override
+            public void run() {
+                synchronized (sparkContextsStates) {
+                    final long currentTime = System.nanoTime();
+                    if (!sparkContextsStates.isEmpty()) {
+                        for (SparkContextState sparkContextState: sparkContextsStates.values()) {
+                            if (!sparkContextState.isLocked() && sparkContextState.hasTimedOut()) {
+                                getLogger().info(String.format("Found context %s which timed out", sparkContextState.ContextName));
+                                deleteContext(sparkContextState.ContextName);
+                            }
+                        }
+                    }
+                }
+            }
+        };
+        long delay = 0;
+        long intevalPeriod = 10000;
+
+        // schedules the task to be run in an interval
+        contextTimeoutTimer.scheduleAtFixedRate(contextTimeoutTask, delay,
+                                  intevalPeriod);
+        getLogger().info("Created and scheduled context timeout task");
+    }
+
+    /**
+     * Stop timer task and kill all running Spark contexts.
+     */
+    @OnShutdown
+    public void shutdown() {
+        contextTimeoutTimer.cancel();
+        for (SparkContextState sparkContextState: sparkContextsStates.values()) {
+            deleteContext(sparkContextState.ContextName);
+        }
+    }
+
+    /**
+     * Checks to see if a Spark Context is new on the Controller Service
+     * @param ContextName the name of the Context
+     * @return true or false
+     */
+    private boolean isNewContext(String ContextName) {
+        boolean newContext = false;
+        synchronized (sparkContextsStates) {
+            try {
+                boolean exit = false;
+                while (true) {
+                    if (!sparkContextsStates.containsKey(ContextName)) {
+                        getLogger().info(String.format("Context: %s is a new Context in Controller service", ContextName));
+                        sparkContextsStates.put(ContextName, new SparkContextState(ContextName));
+                        newContext = true;
+                        break;
+                    } else if (!sparkContextsStates.get(ContextName).isRunning){
+                        getLogger().info(String.format("Context: %s is being created... waiting", ContextName));
+                        sparkContextsStates.wait();
+                    } else {
+                        break;
+                    }
+                }
+            } catch (Exception ex) {
+                getLogger().error(ex.getMessage());
+            }
+        };
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)